### ISSUE TITLE
toggle port-forward option from command line

### ISSUE THREAD

**Issue by piramiday** [pia-foss/desktop#7]
at present, it is trivial to connect to the _fastest-ping region_ via the command line:
```
piactl set region auto
```
which is great.

independently of the GUI settings, it would be awesome if we could invoke a similar command to connect to the _fastest-ping region **which supports port forwarding**_, e.g.:
```
piactl set region auto-port-forward
```
so that it could be invoked from a script.

I understand the same can be accomplished from the GUI, but I do not see the portforwarding as a global configuration -- it's more of a local configuration to be changed at will (outside of the GUI).

I don't want to be connected to a PF region all the time, but I would love to be able to connect to one via a script, if needed.

**Comment by banister** [comment]
the auto region does already respect port-forwarding - if you toggle on PF, the auto region is constrained to those regions which already have PF.

**Comment by piramiday** [comment]
I already wrote:
> independently of the GUI settings

and
> I understand the same can be accomplished from the GUI, but I do not see the portforwarding as a global configuration -- it's more of a local configuration to be changed at will (outside of the GUI).

my goal is to be able to interface with PF and non-PF regions in a _user-independent_ workflow, **without flicking GUI checkboxes**, e.g. in a bash script:
```bash
piactl set region auto-port-forward
echo "do something that requires PF"

piactl set region auto
echo "do something that requires low ping"
```
at present, if I select PF in the GUI, then I'm stuck with PF for all my sessions, and vice versa.

@banister I'm new here -- are you writing as part of the PIA team, or as a regular user?
I think this issue is for the developers, if and when they'll have time to reply or tackle this.

**Comment by banister** [comment]
i'm a developer :)

Sorry i found your original description hard to follow, but your update is much clearer.

Hypothetically, what if we just exposed the "use port-forwarding" checkbox to the CLI? Then you could toggle it on or off before you connect to an auto region?

**Comment by piramiday** [comment]
> i'm a developer :)

haha, great! if you want, you could add `@pia-foss` to your organization. :smile: 

> Sorry i found your original description hard to follow, but your update is much clearer.

I thought my first post was pretty clear, but okay -- I'm glad I got a second chance to explain! :+1: 

> Hypothetically, what if we just exposed the "use port-forwarding" checkbox to the CLI? Then you could toggle it on or off before you connect to an auto region?

something like `piactl set portforward on/off`? sure, that would work!

IMHO, all GUI options should be toggle-able from the CLI anyway. :technologist:

**Comment by banister** [comment]
You can actually make use of the unstable CLI options to achieve what you want today, try the following to toggle port forwarding (use `false` to turn it off):

```
piactl -u applysettings '{"portForward":true}'
```

**Comment by piramiday** [comment]
seems to be working! thanks. :+1: 

if that's okay with you, I'd like to wait for such feature to be documented somewhere before closing this issue.

reference: https://www.privateinternetaccess.com/helpdesk/kb/articles/pia-desktop-command-line-interface

**Comment by JonathonH-PIA** [comment]
This is now possible in 2.2.0 with `piactl set requestportforward <true/false>`!  `piactl --help` lists this in 2.2.0, and we'll update the KB article.
